src/easypost.js

```diff
@@ -221,19 +266,62 @@ export default class EasyPostClient {
       request.auth(this.key);
     }
 
-    if (params !== {} && params !== undefined) {
+    // it would be ideal if this "full url with params" could be gotten from superagent directly,
+    // but it doesn't seem to be possible
+    const url = new URL(urlPath);
+
+    if (params !== undefined) {
       if (method === METHODS.GET || method === METHODS.DELETE) {
         request.query(params);
+        Object.entries(params).forEach(([key, value]) => {
+          url.searchParams.append(key, value);
+        });
       } else {
         request.send(params);
       }
     }
 
+    const baseHooksValue = {
+      method,
+      path: url.toString(),
+      requestBody: request._data,
+      headers: requestHeaders,
+      requestTimestamp: Date.now(),
+      requestUUID: uuid(),
+    };
+
+    if (this.requestHooks.length > 0) {
+      this.requestHooks.forEach((fn) => fn(baseHooksValue));
+    }
```

Comment:
```suggestion
      this.requestHooks.forEach((fn) => fn(baseHooksValue));
```

Nit: If statement is not necessary here

Reply:
👍